llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.cpp

```diff
@@ -344,9 +344,10 @@ unsigned GCNTTIImpl::getMinVectorRegisterBitWidth() const {
 unsigned GCNTTIImpl::getMaximumVF(unsigned ElemWidth, unsigned Opcode) const {
   if (Opcode == Instruction::Load || Opcode == Instruction::Store)
     return 32 * 4 / ElemWidth;
-  return (ElemWidth == 16 && ST->has16BitInsts()) ? 2
-       : (ElemWidth == 32 && ST->hasPackedFP32Ops()) ? 2
-       : 1;
+  return ElemWidth == 8                                ? 4
+         : (ElemWidth == 16 && ST->has16BitInsts())    ? 2
```

Comment:
It doesn't seem right that we will try to vectorize i8s in all cases but not for i16s.

Reply:
Yes I wondered this myself. It seems that this should be unconditional as well. I don't know (because of lack of comments) what the original intent was so I left it untouched as this seems a change which is unrelated to this patch.

Reply:
I would like to keep this patch about i8s only. If we want to relax conditions for other types it can be done in a separate PR.